src/vs/vscode.d.ts

```diff
@@ -1163,7 +1163,7 @@ declare module 'vscode' {
 		 * @param decorationType A decoration type.
 		 * @param rangesOrOptions Either {@link Range ranges} or more detailed {@link DecorationOptions options}.
 		 */
-		setDecorations(decorationType: TextEditorDecorationType, rangesOrOptions: Range[] | DecorationOptions[]): void;
+		setDecorations(decorationType: TextEditorDecorationType, rangesOrOptions: readonly Range[] | readonly DecorationOptions[]): void;
```

Comment:
I'm using `readonly []` instead of `ReadonlyArray` since `readonly[]` has been supported in TypeScript for a while now and is often more legible

Reply:
Yeah, let's go for it an make the same change in other places